src/haddock/libs/libontology.py

```diff
@@ -233,11 +255,226 @@ def remove_missing(self) -> None:
                 if not element.is_present():
                     idxs.append(idx)
 
-        self.output = [value for i, value in enumerate(self.output) if i not in idxs]
+        self.output = [
+            value for i, value in enumerate(self.output)
+            if i not in idxs
+            ]
 
     def __repr__(self) -> str:
         return f"Input: {self.input}{linesep}Output: {self.output}"
 
+    def load_from_input_molecules(
+            self,
+            input_molecules_dir: Path,
+            ) -> None:
+        """Load first molecules at the stat of the workflow.
+
+        Parameters
+        ----------
+        input_molecules_dir : Path
+            Directory where the input molecules are stored.
+        """
+        # Gather all input molecules
+        input_molecules = list(input_molecules_dir.glob('*.pdb'))
+        assert input_molecules != [], \
+            f"No molecules could be found in `{input_molecules_dir}`"
+        # Sort them by creation date (which is also input order)
+        input_molecules.sort(key=getmtime)  # FIXME: getctime ?
+        # Set input attribute
+        self.input = input_molecules
+
+        # Set parsing variables
+        molecules_list: list[dict[int, PDBFile]] = [
+            Molecule(input_file).pdb_files
+            for input_file in self.input
```

Comment:
`input_file` could be either a `dict` or a `PDBFile` - this will fail if it's not a dict

Reply:
In this case, `input_file` is a `Path` to the input molecules from the config file.